### ISSUE TITLE
Page refreshes or redirects without warning (within 20 hours) - emergency.uiowa.edu

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa801#9732]
# Motivation

Siteimprove is reporting "Page refreshes or redirects without warning (within 20 hours)". This is a WCAG 2.2? A compliance issue related to the Emergency website.

# Siteimprove info

**Technical doc**
https://alfa.siteimprove.com/rules/sia-r9

Siteimprove link to existing issue
- https://my2.siteimprove.com/Inspector/940344/A11y/Page?pageId=47302581264&impmd=264ED37B65FA6186129334A508391970&conf=a+aa#/sia-r9/failed/QkVXdUg3NjY=;NjgzMjU4YjQ=

Example
- https://emergency.uiowa.edu/

**Page refreshes or redirects without warning (within 20 hours)**
Issue
2.2.1: Timing Adjustable
2.2.4: Interruptions
3.2.5: Change on Request
A <meta> element is being used to refresh or redirect visitors away from the page within 20 hours after content has loaded.

**Comment by briand44** [comment]
## Grooming
 
**Quadrant:** 

Complex

 **Discussion:**
 - This was a quick solution to make sure folks were getting view content updated, refresh reloads the page to fetch new content
   - set to 55 since import happening every minute
 - Could do something like Signage reload
 - Using js, we can properly notify, which meta refresh isn't doing
 - Could rewrite view on homepage to be a js block (campus status and recent hawk alerts)
   - https://github.com/uiowa/uiowa/issues/4651
   - Learnings here could have big impact on other blocks
 - Should investigate rule - https://alfa.siteimprove.com/rules/sia-r9

**Proposed solution or definition of done:**

**Comment by GaryRidgway** [comment]
Today (04/13/2026) I started by doing some reporting for MJ off the manifest to hopefully jump start the reports he needs.

Then I filled out my AI gateway pilot survey.

After that, I've been investigating how to do ajax refreshes on blocks, and investigated two modules, both with issues.

The first was [Ajax Block](https://www.drupal.org/project/block_ajax), which sadly only works for the block layout, not layout builder, so that was a dud.

The second is [Views Auto Refresh](https://www.drupal.org/project/views_auto_refresh), which seems promising, but I can't seem to get it to work just quite yet. still investigating settings needed to make it kick into gear.

**Comment by joewhitsitt** [comment]
Short of rebuilding the block, views_auto_refresh looks like a decent drop-in solution. If we are just doing this for one or two views, maybe we can harvest the good stuff and just run with some local code.

**Comment by GaryRidgway** [comment]
For context, we are pivoting to use a js block instead of views refresh block. [This conversation helps understand some of the context.](https://iowaweb.slack.com/archives/C0164FNN1PV/p1776265993715719)

**Comment by bspeare** [comment]
After completing this, what do folks think we could have done better in grooming an issue like this?  It looks like this started as a medium.

**Comment by joewhitsitt** [comment]
Good question to ask. I am only one opinion on this but I think given the knowns, I think it was sized appropriately at the time, but it did not describe what the end result should include, only that JS was the way to do it. I was unaware that situation updates were included with the view.

There was also confusion early on with "what we recently did for facilities" and what the emergency website already provided.

We probably should have re-groomed when we started to discover the additional complexities (situation updates, sharing JS,  different render locations/behavior). Sharing some JS from uiowa_alerts made sense, but it got messy quick. We would have had a duplicative call and similar render, but keeping the JS separate for the slice would have made it simpler.
